### ISSUE TITLE
(More) flexible components

### ISSUE THREAD

**Issue by felixheidecke** [owncloud/owncloud-design-system#219]
@DeepDiver1975 @LukasHirt I propose a flexible dropdown solution for the current implementation of the oc-menu.

Split this into a simple drop-down components with a single slot where multiple items can fit. e.g.:

```
<oc-button-row>
   <oc-button id="foo">I so happen to be placed in a button row</oc-button>
   <oc-button>Hello there</oc-button>
</oc-button-row>

[…]

<oc-dropdown uk-dropdown="boundary: #foo; toggle: #foo">
   <span>some text</span>
   <oc-menu>
      <oc-menu-item />
      <oc-menu-item />
      <oc-menu-item />
   </oc-menu>
   <span>other text</span>
   <oc-search-bar />
</oc-dropdown>
```

This approach allows for easy arrangement on elements in the dropdown, like having an aditional search-bar in the filter-dropdown. The trigger can also live outside in a button-row.

**Comment by felixheidecke** [comment]
The above is an approach on how to write components in general. Less restrictive, more open.

**Comment by DeepDiver1975** [comment]
implemented partially with #226 

do you want to add oc-button-row? @felixheidecke

**Comment by DeepDiver1975** [comment]
has been addressed -> close
